### ISSUE TITLE
Exported rotation incorrect when objects are directly parented to bones

### ISSUE THREAD

**Issue by mjdave** [KhronosGroup/glTF-Blender-IO#603]
**Describe the bug**
When exporting, if you have top level objects parented to bones, the exported rotation is incorrect.

**To Reproduce**
Create an Armature with one or more bones. Add an empty object to the scene, set its parent to one of the bones. Rotate the bones and object to some non-identity value, and export. The rotation in the GLTF file is not useful.

**.blend file/ .gltf**
[blenderExportBug.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/3454869/blenderExportBug.zip)

**Version**
Windows 10, Blender 2.8 initial release

**Additional context**
The problem is in gltf2_blender_gather_nodes.py __gather_children, where it does the "#fix rotation" thing. It doesn't make any sense to me why it multiplies with that different than normal basis change matrix and then the parent inverse. I have got the expected result here by not multiplying with the parent matrix, instead commenting out that if gltf2_blender_export_keys.YUP statement and exporting the unmodified child_node.rotation. Then I multiply that rotation by the standard basis change matrix in my loader. I've so far been unsuccessful in my attempts to multiply by the correct basis change matrix in the exporter, I'm not sure what the problem is there, I would expect the following to work in exactly the same way as my solution, but it doesn't:
```
rot_mat = Quaternion(rot).to_matrix().to_4x4()

axis_basis_change = Matrix(
    ((1.0, 0.0, 0.0, 0.0), (0.0, 0.0, 1.0, 0.0), (0.0, -1.0, 0.0, 0.0), (0.0, 0.0, 0.0, 1.0)))
mat = gltf2_blender_math.multiply(axis_basis_change, rot_mat)
```

**Comment by aguaviva** [comment]
I think the issue is with the way YUP is implemented.

Given a vector 'v' and a matrix 'm' in Blender's original base, to get them in the YUP base we need to transform them this way:
1) a vector v should become (axis_basis_change * v)
2) a matrix m should become (axis_basis_change * axis_basis_change * axis_basis_change^-1)

And then all will work as expected.

So you'd need to add this at the bottom of your code:
> mat = gltf2_blender_math.multiply(mat, inverse(axis_basis_change) )

Cameras and lights suffer from a similar issue and would need the same treatment.

**Comment by aguaviva** [comment]
@mjdave do you get the right results when you are not using checking "+Y UP" option?

**Comment by aguaviva** [comment]
@mjdave do you get the right results when you are not using checking "+Y UP" option?

Also @mjdave the matrix to change from 'Blender' into 'YUP' spaces seems to be defined in 5 different places, i'd be nice to have this matrix and all the functions for changing into the YUP space into just one place.

Also I would try to do all the math in 'Blender space' and do the transformation into 'YUP space' at the very end, when you are writing it to the json structure.

**Comment by julienduroure** [comment]
Please provide a test case, not with Empty, so we will be able to SEE the bug directly in any external viewer like https://gltf-viewer.donmccurdy.com/
I was not able to reproduce with simple case, so I want to show your test case.
Thanks !

**Comment by mjdave** [comment]
My apologies. I've attached a test case that illustrates the problem clearly. If you ensure the +Y UP option is selected and export the model, then the monkey face will be facing the wrong direction relative to the cone when viewed in https://gltf-viewer.donmccurdy.com/ whereas without that option selected, the result is correct.
[testCase2.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/3464505/testCase2.zip)

**Comment by julienduroure** [comment]
I can confirm the bug

**Comment by julienduroure** [comment]
@donmccurdy  Interesting, you don't have the save visual result in https://gltf-viewer.donmccurdy.com/ and in https://sandbox.babylonjs.com/ 
Results seems better (good?) with babylonjs
I produced 2 new .blend files with corresponding .glb to better visualize the rotation. (the second one is without initial rotation on suzanne head, in order to check if this have an impact)
[bug_603.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/3464573/bug_603.zip)

**Comment by donmccurdy** [comment]
If I export with the *Always sample animations* option enabled, the bug seems to go away. Based on that, my guess is that either there's a bug in the threejs handling of cubicspline in/out tangents, or (less likely) the in/out tangents are incorrect and babylonjs is just not using them. I used to have a test model to check if an engine implemented the in/out tangents but can't seem to find it at the moment.

Will have to look at this a bit more, meanwhile sampling animations may be a workaround.

**Comment by julienduroure** [comment]
Note: this issue will be managed only after #566
